### PR TITLE
Added re-usable toolbar

### PR DESCRIPTION
**Pull request by hferrier**

### CODE REVIEW

File: src/view/Overview.tsx [hunk 1]
```diff
@@ -4,25 +4,44 @@ import { observer } from 'mobx-react';
...
+        specifications={state.specificationList}
+        expandedSpecificationId={state.expandedSpecificationId}
+        // Expands/collapses a specification
+        onSpecificationExpanded={id => (state.expandedSpecificationId = id)}
+        // Go to the specification viewing route when you select a specification
+        onSpecificationSelected={specification =>
+          history.push(`/specifications/${specification.id}`)
+        }
+        // Opens the 'Add Specification' modal
+        onAddSpecificationModalOpened={() => history.push(`${match.url}/add`)}
```
Comment (calebj0seph) [inline src/view/Overview.tsx]: Can remove this and just make it work with your new toolbar button :)

File: src/view/ProfileViewer.tsx [hunk 2]
```diff
@@ -3,12 +3,30 @@ import { observer } from 'mobx-react';
...
+
 // TODO: Add react-router's injected props
 export const ProfileViewer: SFC<any> = observer(() => (
-  <ContentContainer>
-    {
-      // Using state.me for now as example data
-    }
-    <ProfileInformation profile={state.me} />
-  </ContentContainer>
+  <span>
```
Comment (calebj0seph) [inline src/view/ProfileViewer.tsx]: Same here, think a `div` makes more sense.

File: src/view/basic/SimpleToolbar.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+          placeholder={searchPrompt}
+          className={classes.searchArea}
+          InputProps={{
+            disableUnderline: true,
+            endAdornment: <SearchIcon className={classes.searchIcon} />,
+            className: classes.searchInput
+          }}
+          onChange={(event: any) => onSearchInputChange(event.target.value)}
+        />
+        <section className={classes.actions}>{actions.map(action => action)}</section>
```
Comment (calebj0seph) [inline src/view/basic/SimpleToolbar.tsx]: Is `actions.map(action => action)` needed? Can't you just do `actions`?

File: src/view/NavigationMenu.tsx [hunk 4]
```diff
@@ -75,24 +82,35 @@ export const NavigationMenu: SFC<NavigationMenuProps> = () => (
                 <List component="nav">
                   <NavigationButton
                     onClick={() => navState.toggleOpen()}
-                    icon={navState.open ? <Icons.ChevronLeft /> : <Icons.ChevronRight />}
+                    icon={
+                      navState.open ? (
+                        <Icons.ChevronLeft className={classes.navIcon} />
```
Comment (PatrickShaw) [inline src/view/NavigationMenu.tsx]: I don't think you need the ()

File: src/view/ProfileViewer.tsx (hunk 2 above)
Comment (PatrickShaw) [reply src/view/ProfileViewer.tsx]: Yeah, or header.

**Comment by PatrickShaw** [comment]
Could you add a Storybook story for this? Pull `development` and have a look at how I did it for `SpecificiationModal`.

File: src/view/ProfileViewer.tsx (hunk 2 above)
Comment (hferrier) [reply src/view/ProfileViewer.tsx]: Yeah, this is mostly just a hack to get around the fact that the observer HOC won't accept any array of JSX elements like the regular constructor would. I'll make it a div for now.

**Comment by calebj0seph** [comment]
Looks good now :D

**Pull request merged**